File: 260-single-number-iii/single-number-iii.cpp
class Solution {
public:
    vector<int> singleNumber(vector<int>& nums) {
       int mask = 0;
       for(auto &x : nums) mask ^= x;
       int a = 0, b = 0;
       int diffBit = 1;
        while ((mask & diffBit) == 0) {
            diffBit <<= 1;
        }
       for(auto &x : nums){
        if(x  & diffBit ) a ^= x;
        else b ^= x;
       }
       return {a,b}; 
    }
};

File: 260-single-number-iii/single-number-iii.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var singleNumber = function(nums) {
   const mp = {};
   for(let i = 0 ; i < nums.length ; i++){
    if(mp.hasOwnProperty(nums[i])){
        mp[nums[i]] = mp[nums[i]] + 1;
    }else{
       mp[nums[i]] = 1;
    }
   } 
    let ans = [];
   for(const [key,value] of Object.entries(mp)){
    if(value == 1){
        ans.push(key);
    }
    console.log(key,value);
   }
   return ans;
};